### PR TITLE
Fix nullability warnings in System.Diagnostics.Process

### PR DESCRIPTION
**Pull request by stephentoub**
Two PRs (one to nullable-annotate S.D.Process, and one to add a method to the assembly) raced in CI.
cc: @safern, @AndyAyersMS

### CODE REVIEW

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs [hunk 1]
```diff
@@ -1426,9 +1426,9 @@ public async Task WaitForExitAsync(CancellationToken cancellationToken = default
                 throw;
             }
 
-            var tcs = new TaskCompletionSourceWithCancellation<object>();
+            var tcs = new TaskCompletionSourceWithCancellation<bool>();
```
Comment (safern) [inline src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs]: Out of curiosity why was this needed?

**Review by safern — APPROVED** [review]

File: src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs]: It needed to be made `object?` so that the `null` passed to `TrySetResult` was valid.  And as long as I was changing it, I changed it to just be a bool, which is what we generally use elsewhere in such situations where the T is actually just a dummy value.

Comment (safern) [reply src/libraries/System.Diagnostics.Process/src/System/Diagnostics/Process.cs]: I see. Makes sense. Thanks for explaining.

**Comment by safern** [comment]
Merging since failures are in other PRs and builds and to unblock other failures on CI.
